### ISSUE TITLE
Fix `plugin-site` chart issues: `github.appId is required` & `jira.username is required`

### ISSUE THREAD

**Issue by lemeurherve** [jenkins-infra/helm-charts#660]
`helm template plugin-site` output:

> Error: execution error at (plugin-site/templates/secret.yaml:10:20): github.appId is required

> Error: execution error at (plugin-site-issues/templates/secret.yaml:11:20): jira.username is required

Unit tests should be added, doc: https://github.com/helm-unittest/helm-unittest/blob/main/DOCUMENT.md

**Comment by bhavyastar** [comment]
Hey, I would like to work on this issue.

**Comment by bhavyastar** [comment]
Yes, I had a look at all of these issues. I understand the issue, and after making a draft pull request you can tell me if there are necessary changes that need to be made. Is that okay?

**Comment by ashwin1596** [comment]
Hi,

I am interested in working on this issue if it is still open and a PR hasn't already been raised. Could you please confirm if I can start working on it?

Thank you!

**Comment by dduportal** [comment]
> Hi,
> 
> I am interested in working on this issue if it is still open and a PR hasn't already been raised. Could you please confirm if I can start working on it?
> 
> Thank you!

yes you can. Don't forget to check https://github.com/jenkins-infra/helm-charts/issues/725 before

**Comment by ashwin1596** [comment]
@dduportal, I reviewed ticket #725, where the problem was that the values were initially null but needed to be strings. This was resolved by setting them to "". However, the current error indicates that these fields are still mandatory despite containing "". Should we ensure they have non-empty strings, or is there a specific adjustment needed in secret.yaml? Could you advise on the next steps?

**Comment by PalashChitnavis** [comment]
hey @dduportal, I would like to work on this, is the issue still open?

**Comment by ParkarAbhi09** [comment]
Hello,
While analyzing previous conversations and issue #725, I found that the `accountapp` is responsible for account management. The error occurred because Helm expects all secret values to be encoded as base64 strings, which require valid input. If the value was `null`, the` b64enc` function failed as it cannot encode `null`. This caused the Helm template rendering process to break.

Fix Implemented
To resolve this, the password was set to an empty string (`""`). This ensures that Helm does not fail on `b64enc` if no password is provided. Helm will still encode an empty string without error, effectively fixing the Helm template for `accountapp`.

Issue with `plugin-site` (#660)
However, if we try to deploy the Helm template for `plugin-site`, an error will occur because certain fields are required, unlike in `accountapp`. The `plugin-site` is responsible for displaying plugin-related details, fetching updates, new plugin information, source code, and issues from GitHub and Jira.

Since `plugin-site` requires Jira and GitHub credentials to fetch plugin data, missing values will not cause deployment failure, but the app will be unable to retrieve or display data. It will still attempt to call the GitHub/Jira APIs, which will reject unauthenticated requests, leading to errors like `403` or `401`.

Unit Test Implementation
I also reviewed the existing unit test for `accountapp` and followed a similar approach while writing the unit tests. The unit tests were written in a separate file to improve code readability. The test cases cover:

When `githubId` is missing.

When `jiraUsername` is missing.

When all required values are available.

Conclusion
The Helm template for `plugin-site` will still fail due to required fields, and removing them is not the correct approach for production deployments. The unit tests have been added in a separate file as well, following the structure of the `accountapp` unit tests.

Would you like this fix, or do you need any additional changes?

**Comment by tiya-9975** [comment]
I reviewed the discussion and the recent PR #1598 related to this issue.
To help move this forward, I can test the current PR locally and verify:

- Whether the helm template still fails when github.appId or jira.username are missing
- Whether the unit tests correctly cover the missing-field scenarios
- Whether the chart behaves consistently with issue #725 (as suggested earlier)

If maintainers are okay with it, I can share validation results here.
